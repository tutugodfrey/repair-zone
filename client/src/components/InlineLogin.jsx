import React, {Component} from 'react';
import Form from './elementComponents/Form.jsx';
import FormInput from './elementComponents/FormInput.jsx';
import Button from './elementComponents/Button.jsx';
import dataFiledCollecotor from '../services/dataFieldCollector';

export default class InlineLogin extends Component {
  constructor() {
    super();
    this.state = {}
  }

  getPassword(event) {
    event.preventDefault();
    this.setState({
      password: event.target.value,
    });
  }
  getUsername(event) {
    event.preventDefault();
    this.setState({
      username: event.target.value,
    });
  }
  handleLogin(event) {
    event.preventDefault();
    const password = this.state.password;
    const username = this.state.username;
  }
  formContent() {
    return <div id='home-page-login' className='d-flex flex-row d-sm-none d-md-block'>
      <FormInput
      inputType='text' 
      inputClass='ml-1 form-controls-md' 
      inputId='signin-1-username' 
      inputPlaceholder='username' 
      inputName='username'
      onChange={this.getUsername.bind(this)}/>
      <FormInput
      inputType='text'
      inputClass='ml-1'
      inputId='signin-1-password'
      inputPlaceholder='password'
      inputName='password'
      onChange={this.getPassword.bind(this)} />
      <Button
      buttonClass='ml-1 bg-success text-white' 
      buttonId='' 
      buttonName='log-In'
      onClick={this.handleLogin.bind(this)} />
      </div>
  }
  render() {
    return <Form formId='' formClass='' content={this.formContent()} />
  }
}
