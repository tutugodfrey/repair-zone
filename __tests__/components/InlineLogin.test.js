import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { InlineLogin } from '../../client/src/components/InlineLogin.jsx';

describe('<InlineLogin />', () => {
  describe('unconnected component test', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(
      <InlineLogin
        setFormToFill={jest.fn()}
      />
      );
    });
    test('should map the snapshot of the component', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
