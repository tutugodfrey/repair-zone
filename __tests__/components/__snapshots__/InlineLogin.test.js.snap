// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineLogin /> unconnected component test should map the snapshot of the component 1`] = `
<Form
  content={
    <div
      className="d-none d-md-block"
      id="home-page-login"
    >
      <FormInputWithoutDiv
        inputClass="ml-1 border-0 rounded py-1 px-2 field-valid required"
        inputId="signin-1-username"
        inputName="username"
        inputPlaceholder="username"
        inputType="text"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
      />
      <span
        className="text-danger"
      />
      <FormInputWithoutDiv
        inputClass="ml-1 border-0 rounded py-1 px-2 field-valid required"
        inputId="signin-1-password"
        inputName="password"
        inputPlaceholder="password"
        inputType="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
      />
      <span />
      <Button
        buttonClass="ml-1 bg-success text-white border-0 rounded py-1 px-2"
        buttonId="login-1-btn"
        buttonName="Log-In"
        onClick={[Function]}
      />
    </div>
  }
  formClass=""
  formId="inline-signin-form"
/>
`;
